### DISCUSSION TITLE
Old UI

### DISCUSSION THREAD

**Discussion by RoseStarlight** [RPTools/maptool#3717] [Ideas]
Hello! I really enjoy they new update, themes are much better now, its fantastic! However, I really dislike the new UI. I was wondering if it was possible to add a feature so we could change back to the old style?

**Comment by Phergus** [comment]
Not sure what you mean by "new UI".  It's mostly the same UI with a different theme.  The only thing that changed off the top of my head was the layout of some of the tabs of the Edit Token dialog so that people with DOS-era displays could see the whole thing.

**Comment by RoseStarlight** [comment]
Apologies! I should've clarified. Here is a small comparison between the old style and new. While I can see the appeal, it's just not for me. I really like when the UI pops and has clear boundaries. So that's why I'm hoping there is a possibility of having the option to use the previous style. 
 New ![New](https://user-images.githubusercontent.com/44916049/197874770-0c7c5bda-837e-4577-94d4-4dfe43b2aa65.png) Previous   ![Old](https://user-images.githubusercontent.com/44916049/197874857-da583339-1e3c-4189-ab3b-0a4f489b09eb.png)

**Comment by Phergus** [comment]
I see.  It's a function of the new Look & Feel engine which is what gives us the new theming.  It's possible that a custom theme might be able to give us some of that look back.  I haven't dug into the new theme stuff to see what you can do.

FWIW, I prefer the old style as well but for years we've been hearing how out-dated the interface looked and so of course once we switched to what is the current trend in UI design we now get the opposite reaction.  :)

**Comment by RoseStarlight** [comment]
Yeah after looking into the new theming engine, it looks like its built around being flat and modern, so It would be hard to mimic the old style. Kind of unfortunate hahaha. 
I never really looked into the git hub until this update, so I never saw the requests for an updated UI. There's always going to be people on the other side of the opinion with any changes hahahah. I'll just get used to the new format, Thanks for the replies though!

**Reply by bubblobill** [reply]
I made the theme Aah to be the closest approximation I could make using the new LaF to be like the old non-flat LaF.

**Comment by FullBleed** [comment]
Keep in mind that the new Flat LaF is more than just a "modern" facelift.

Some of the plus-ups include:  It now will allow for "reactive" icons that change color based on the selected theme, it now heeds the OS scaling and hdpi settings throughout the app (which is getting more and more important as people use greater than 1080p displays), there are now 60+ themes to choose from, the dark themes default to *much* more legible colors in various windows, and Flat LaF is in active development for modern Java implementations.

And while I would like some more "3d" type buttons (i.e. gradients and shadows), I wouldn't go back to all of those short comings to get it.  There are a couple themes in Flat Laf that have some buttons with gradients (Space gray is one)... so it might be possible to edit some existing themes to have more gradients... no one has done it yet though.
